src/librustc/middle/const_eval.rs

```diff
@@ -877,12 +962,57 @@ pub fn eval_const_expr_partial<'tcx>(tcx: &ty::ctxt<'tcx>,
       ast::ExprBlock(ref block) => {
         match block.expr {
             Some(ref expr) => try!(eval_const_expr_partial(tcx, &**expr, ety)),
-            None => const_int(0)
+            None => const_int(0) // huh???
```

Comment:
oh sorry, should have been researching better: https://github.com/rust-lang/rust/pull/14183#discussion_r12612731